File: jvm/src/main/java/it/unibo/testlecture/u03_coverage/ProgramToCover.java
package it.unibo.testlecture.u03_coverage;

public class ProgramToCover {
  public static void methodToCover(boolean c1, boolean c2, boolean c3, boolean c4, boolean c5, int k) {
    System.out.print("A");
    if(c1 && c2) System.out.print("B"); else System.out.print("C");
    System.out.print("D");
    if(c3 || c4) System.out.print("E"); else System.out.print("F");
    System.out.print("G");
    if(c5) System.out.print("H");
    for(int j = 1; j <= k; j++) System.out.print(j);
  }
}

File: dotnet/u02-unit-testing/it/unibo/testlecture/u03_coverage/ProgramToCover.cs
using System;

namespace it.unibo.testlecture.u03_coverage {
    public class ProgramToCover {
        public static void methodToCover(bool c1, bool c2, bool c3, bool c4, bool c5, int k) {
            Console.Write("A");
            if(c1 && c2) Console.Write("B"); else Console.Write("C");
            Console.Write("D");
            if(c3 || c4) Console.Write("E"); else Console.Write("F");
            Console.Write("G");
            if(c5) Console.Write("H");
            Console.WriteLine();
            // for(int j = 1; j <= k; j++) Console.Write(j);
        }
    }
}